src/libraries/System.Private.Xml/src/System/Xml/Xsl/XPathConvert.cs

```diff
@@ -19,3178 +10,14 @@ namespace System.Xml.Xsl
-                    }
-                    biDbl.InitFromDigits(rgu0, rgu1, 1 + (0 != rgu1 ? 1 : 0));
-                    if (c5Dbl > 0)
-                    {
-                        biDbl.MulPow5(c5Dbl);
-                    }
-                    if (c2Dbl > 0)
-                    {
-                        biDbl.ShiftLeft(c2Dbl);
-                    }
-
-                    wT = biDbl.CompareTo(biDec);
-                    if (wT > 0 || 0 == wT && 0 != (DblLo(dbl) & 1))
-                    {
-                        // Return the next lower value.
-                        dbl = BitConverter.Int64BitsToDouble(BitConverter.DoubleToInt64Bits(dbl) - 1);
-                    }
-                }
-                else
-                {
-                    // biDbl is smaller. Recompute with the dbl plus half the distance
-                    // to the next larger double.
-                    if (0 != AddU(ref rgu0, wAddHi))
-                    {
-                        AddU(ref rgu1, 1);
-                    }
-                    biDbl.InitFromDigits(rgu0, rgu1, 1 + (0 != rgu1 ? 1 : 0));
-                    if (c5Dbl > 0)
-                    {
-                        biDbl.MulPow5(c5Dbl);
-                    }
-                    if (c2Dbl > 0)
-                    {
-                        biDbl.ShiftLeft(c2Dbl);
-                    }
-
-                    wT = biDbl.CompareTo(biDec);
-                    if (wT < 0 || 0 == wT && 0 != (DblLo(dbl) & 1))
-                    {
-                        // Return the next higher value.
-                        dbl = BitConverter.Int64BitsToDouble(BitConverter.DoubleToInt64Bits(dbl) + 1);
-                    }
-                }
-                return dbl;
-            }
-#endif
-
-            private void InitFromDouble(double dbl)
-            {
-                if (0.0 == dbl || IsSpecial(dbl))
-                {
-                    _exponent = 0;
-                    _sign = 1;
-                    _mantissaSize = 0;
-                }
-                else
-                {
-                    // Handle the sign.
-                    if (dbl < 0)
-                    {
-                        _sign = -1;
-                        dbl = -dbl;
-                    }
-                    else
-                    {
-                        _sign = 1;
-                    }
-
-                    if (!BigNumber.DblToRgbFast(dbl, _mantissa, out _exponent, out _mantissaSize))
-                    {
-                        BigNumber.DblToRgbPrecise(dbl, _mantissa, out _exponent, out _mantissaSize);
-                    }
-                }
-            }
-        };
-
-        /*  ----------------------------------------------------------------------------
-            DoubleToString()
-
-            Converts a floating point number to a string according to XPath rules.
-        */
-        public static string DoubleToString(double dbl)
```

Comment:
I'll need to see some test cases making sure we return the same as we did before for all edge cases here. I do not feel confident `double.ToString(CultureInfo.InvarianCulture)` returns the same as this method.

Reply:
I haven't looked deeply into the code yet, but the behavior looks to be changed. The old code doesn't seem to output `e` notations. But I'm also not sure what's the desired behavior.

Reply:
That's why I prefer to split this change. We should document the changes in the issue and see what the spec says vs what is happening right now. If current behavior is not matching the spec and new one is then we can change it but let's have the issue first so we can see clearly what's happening and document why the change happens